data/migrations/20190110201359_instructions_table.js

```diff
@@ -0,0 +1,15 @@
+
+exports.up = function(knex, Promise) {
+    return knex.schema.createTable('instructions', table => {
+        table.increments();
+        table.integer('step_number').notNullable();
+        table.string('step').notNullable();
+        table.integer('recipes_id').unsigned();
+        table.foreign('recipes_id').references('id').on('recipes')
```

Comment:
There's a good amount of repetition with the creation of these tables. For example, what if we broke that out and used a helper function for your foreign keys? For example, have this stored in a separate file:
```javascript
module.exports = {
  foreign: (table, ref) => {
    const fKey = `${ref}_id`;
    table.integer(fKey).unsigned();
    table.foreign(fKey).references('id').on(ref);
  }
}; 
```
and then the usage in this migration would be:
```javascript
const { foreign } = require('./path_to_helper')

exports.up = function(knex, Promise) {
    return knex.schema.createTable('instructions', table => {
        table.increments();
        table.integer('step_number').notNullable();
        table.string('step').notNullable();
        foreign(table, 'recipes');
    })
};
```